### DISCUSSION TITLE
Adding the PR number to commit messages

### DISCUSSION THREAD

**Discussion by rossjrw** [community/community#23650] [New to GitHub]
<p>I’ve seen a lot of repositories where every single commit is suffixed with the respective PR number in parentheses. I’ve decided that I’d like to do this in my own repositories.</p>
<p>I’d assumed that the merge commit message was just being changed manually by whoever is responsible for merging, but on inspecting things more closely, that doesn’t appear to be the case. Merge commit messages aren’t visible in Github’s code browser, but every file has the PR number attached. Here’s github/linguist as an example:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/3/a3ffcf724a954587027adff909a9f1b71ba40dcc.png" data-download-href="/uploads/short-url/noNTJ0aqtBFqgRcxkAojzYfLGc4.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29130152/181078642-1afd707b-240d-4641-a5f1-9411e2f50211.png" alt="image" data-base62-sha1="noNTJ0aqtBFqgRcxkAojzYfLGc4" width="690" height="210" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3ffcf724a954587027adff909a9f1b71ba40dcc_2_690x210.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3ffcf724a954587027adff909a9f1b71ba40dcc_2_1035x315.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/3/a3ffcf724a954587027adff909a9f1b71ba40dcc.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3ffcf724a954587027adff909a9f1b71ba40dcc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1153×351 26.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>My next thought was that the each commit simply has the PR number in the commit message. But that’s stupid and an unreasonable demand for contributors. Looking at the commits in each PR, they don’t feature the PR number at all - not even the merge commit. Here’s an example from github/linguist: <a href="https://github.com/github/linguist/pull/5081/commits" rel="noopener nofollow ugc">https://github.com/github/linguist/pull/5081/commits</a></p>
<p>I can see that the merge was made on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>, so there’s not any fancy command-line tools at play here. This should be achievable via the web interface - right? And it’s not a Github Action, either, given that the only workflow in github/linguist is CI.</p>
<p>How do I automatically and reliably add PR numbers to commits?</p>

**Comment by kingthorin** [comment, marked as answer]
<p>I believe this happens when you use the “Squash and merge” option in the web UI.</p>
